### PR TITLE
[Feature] Dispatch SAC loss module

### PR DESCRIPTION
**Pull request by Blonck**
## Description

Enable dispatching arguments for the `.forward()` method of the SAC loss module.

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: torchrl/objectives/sac.py [hunk 1]
```diff
@@ -88,6 +88,112 @@ class SACLoss(LossModule):
...
+                loss_qvalue: Tensor(shape=torch.Size([]), device=cpu, dtype=torch.float32, is_shared=False),
+                loss_value: Tensor(shape=torch.Size([]), device=cpu, dtype=torch.float32, is_shared=False)},
+            batch_size=torch.Size([]),
+            device=None,
+            is_shared=False)
+
+    This class is compatible with non-tensordict based modules too and can be
+    used without recurring to any tensordict-related primitive. In this case,
+    the expected keyword arguments are:
+    ``["action", "next_reward", "next_done"]`` + in_keys of the actor, value, and qvalue network
```
Comment (vmoens) [inline torchrl/objectives/sac.py:150]:
```suggestion
    ``["action", "next_reward", "next_done"]`` + in_keys of the actor, value, and qvalue network.
```

File: torchrl/objectives/sac.py [hunk 2]
```diff
@@ -88,6 +88,112 @@ class SACLoss(LossModule):
...
+        ...     in_keys=["observation"])
+        >>> loss = SACLoss(actor, qvalue, value)
+        >>> batch = [2, ]
+        >>> action = spec.rand(batch)
+        >>> loss_actor, loss_qvlaue, _, _, _, _ = loss(
+        ...     observation=torch.randn(*batch, n_obs),
+        ...     action=action,
+        ...     next_done=torch.zeros(*batch, 1, dtype=torch.bool),
+        ...     next_observation=torch.zeros(*batch, n_obs),
+        ...     next_reward=torch.randn(*batch, 1))
```
Comment (vmoens) [inline torchrl/objectives/sac.py:194]: For modules that inherit from TensorDIctModuleBase, you can call `module.select_out_keys("loss_actor", "loss_qvalue")` and you should only get those two values out

**Review by vmoens — APPROVED** [review]
Wonderful! Some very minor comments

File: torchrl/objectives/sac.py (hunk 2 above)
Comment (Blonck) [reply torchrl/objectives/sac.py:194]: But this is the example for the @dispatch which returns a list.

Comment (vmoens) [reply torchrl/objectives/sac.py:194]: my point was merely that you could avoid the `_, _, _` by calling `select_out_keys` after constructing the loss

Comment (Blonck) [reply torchrl/objectives/sac.py:194]: Got it. But the loss modules are derived from LossModule which itself is derived from nn.Module, so no `select_out_keys`.

Comment (vmoens) [reply torchrl/objectives/sac.py:194]: Oh I thought we made it inherit from TDMBase, my mistake
